app/models/chargeback_vm.rb

```diff
@@ -98,8 +98,8 @@ def self.where_clause(records, options, region)
   end
 
   def self.extra_resources_without_rollups(region)
-    # support hyper-v for which we do not collect metrics yet (also when we are including metrics in calculations)
-    scope = @options.include_metrics? ? ManageIQ::Providers::Microsoft::InfraManager::Vm : vms(region)
+    # support VM types for which we do not collect metrics yet (also when we are including metrics in calculations)
+    scope = @options.include_metrics? ? Vm.where.not(:type => Vm.types_supporting(:capture)) : vms(region)
```

Comment:
NOTE we need an AR relation here since we use it as a scope later so we can't use the instance-level checks that e.g. `Vm.supporting(:capture)` would do since it returns an array.

Reply:
Hmmm...Can you include your GitHub comment as a comment in the actual code? I can see someone in the future "fixing" this to use `Vm.supporting(:capture)`, not knowing this detail.

This also makes me wonder if we should make a `Vm.possibly_supporting(:capture)` or `Vm.potentially_supporting(:capture)` (or something like that) that does this but returns a scope.

Reply:
Only waiting on this comment before merge,

Reply:
:+1: added above

Reply:
> This also makes me wonder if we should make a Vm.possibly_supporting(:capture) or Vm.potentially_supporting(:capture) (or something like that) that does this but returns a scope.

Yeah I was working up in another PR `.type_supporting`, `.type_not_supporting`, and `not_supporting` where the first two do class-level checks only.  Until we get supports to be tracked in the database there really isn't a great way to do performant per-instance checks so I like to filter by type then check individual instances as much as possible.